snuba/utils/streams/batching.py

```diff
@@ -25,29 +27,33 @@
 
 
 class AbstractBatchWorker(ABC, Generic[TPayload, TResult]):
-    """The `BatchingConsumer` requires an instance of this class to
+    """
+    The ``BatchProcessingStrategy`` requires an instance of this class to
     handle user provided work such as processing raw messages and flushing
-    processed batches to a custom backend."""
+    processed batches to a custom backend.
+    """
 
     @abstractmethod
     def process_message(self, message: Message[TPayload]) -> Optional[TResult]:
-        """Called with each raw message, allowing the worker to do
-        incremental (preferably local!) work on events. The object returned
-        is put into the batch maintained by the `BatchingConsumer`.
+        """
+        Called with each raw message, allowing the worker to do incremental
+        (preferably local!) work on events. The object returned is put into
```

Comment:
Curiosity. Do we actually have a reason to prefer local work in the batching consumer or does it just smells bad.

Reply:
I think the assumption here is that non-local work is slower than local work, which would be likely be (but is not guaranteed to be) problematic since this method is called synchronously.